src/traces/scattergl/convert.js

```diff
@@ -0,0 +1,397 @@
+/**
+* Copyright 2012-2018, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var svgSdf = require('svg-path-sdf');
+var rgba = require('color-normalize');
+
+var Registry = require('../../registry');
+var Lib = require('../../lib');
+var Drawing = require('../../components/drawing');
+var AxisIDs = require('../../plots/cartesian/axis_ids');
+
+var formatColor = require('../../lib/gl_format_color');
+var subTypes = require('../scatter/subtypes');
+var makeBubbleSizeFn = require('../scatter/make_bubble_size_func');
+
+var constants = require('./constants');
+
+function convertStyle(gd, trace) {
+    var i;
+
+    var opts = {
+        marker: null,
+        line: null,
+        fill: null,
+        errorX: null,
+        errorY: null,
+        selected: null,
+        unselected: null
+    };
+
+    if(trace.visible !== true) return opts;
+
+    if(subTypes.hasMarkers(trace)) {
+        opts.marker = convertMarkerStyle(trace);
+        opts.selected = convertMarkerSelection(trace, trace.selected);
```

Comment:
In some reason `trace.selected` is empty here for multiple colors/multiple opacities #2500

Reply:
Ok - I'll have a look at this tomorrow